zcash_client_backend/src/decrypt.rs

```diff
@@ -47,12 +47,12 @@ pub struct DecryptedOutput<Note> {
 
 /// Scans a [`Transaction`] for any information that can be decrypted by the set of
 /// [`UnifiedFullViewingKey`]s.
-pub fn decrypt_transaction<P: consensus::Parameters>(
+pub fn decrypt_transaction<P: consensus::Parameters, A: Copy>(
```

Comment:
It would be more general to have this bound as `Clone`

Reply:
Clone doesn't appear to derive from Copy. Are you saying I should _add_ `Clone` or replace `Copy` with `Clone`? If replace, I'd like to learn why avoiding `Copy` is desirable.

Reply:
`Copy` derives from `Clone`

Reply:
How do I get the FnMut closure to work given it wants to Copy the `account` variable? Adding `.clone()` to it within the lamba (of course) doesn't work.

Reply:
Essentially, you need the clone in two places: ```diff --git a/zcash_client_backend/src/decrypt.rs b/zcash_client_backend/src/decrypt.rs
index 1d7231306..5918441cd 100644
--- a/zcash_client_backend/src/decrypt.rs
+++ b/zcash_client_backend/src/decrypt.rs
@@ -47,7 +47,7 @@ pub struct DecryptedOutput<Note, AccountId> {

 /// Scans a [`Transaction`] for any information that can be decrypted by the set of
 /// [`UnifiedFullViewingKey`]s.
-pub fn decrypt_transaction<P: consensus::Parameters, A: Copy>(
+pub fn decrypt_transaction<P: consensus::Parameters, A: Clone>(
     params: &P,
     height: BlockHeight,
     tx: &Transaction,
@@ -74,6 +74,7 @@ pub fn decrypt_transaction<P: consensus::Parameters, A: Copy>(
                         .iter()
                         .enumerate()
                         .flat_map(move |(index, output)| {
+                            let account = account.clone();
                             try_sapling_note_decryption(&ivk_external, output, zip212_enforcement)
                                 .map(|ret| (ret, TransferType::Incoming))
                                 .or_else(|| {
@@ -92,7 +93,7 @@ pub fn decrypt_transaction<P: consensus::Parameters, A: Copy>(
                                 .map(move |((note, _, memo), transfer_type)| DecryptedOutput {
                                     index,
                                     note,
-                                    account,
+                                    account: account.clone(),
                                     memo: MemoBytes::from_bytes(&memo).expect("correct length"),
                                     transfer_type,
                                 })

```